processing/src/main/java/io/druid/segment/StringDimensionIndexer.java

```diff
@@ -534,79 +535,6 @@ public void fillBitmapsFromUnsortedEncodedArray(
     }
   }
 
-  @Override
-  public ValueMatcher makeIndexingValueMatcher(
-      final String matchValue,
-      final IncrementalIndexStorageAdapter.EntryHolder holder,
-      final int dimIndex
-  )
-  {
-    final int encodedVal = getEncodedValue(matchValue, false);
-    final boolean matchOnNull = Strings.isNullOrEmpty(matchValue);
-    if (encodedVal < 0 && !matchOnNull) {
-      return new BooleanValueMatcher(false);
-    }
-
-    return new ValueMatcher()
```

Comment:
I would really like to be able to see this happen in a benchmark, since otherwise it's challenging to prevent performance regressions. A future patch should be adding more kinds of ValueMatcherColumnSelectorStrategies. Do you think it'd work to have two different types of value matchers used in the same benchmark? We could do that in FilteredAggregatorBenchmark.

I'll file an issue for 0.10.0 to look into this.